ovs/extensions/db/arakoon/ArakoonInstaller.py

```diff
@@ -244,19 +319,44 @@ def clean_leftover_arakoon_data(ip, directories):
             root_client.file_delete(info['files'])
 
     @staticmethod
-    def create_cluster(cluster_name, ip, base_dir, plugins=None, locked=True):
+    def create_cluster(cluster_name, cluster_type, ip, base_dir, plugins=None, locked=True, internal=True):
         """
-        Creates a cluster
-        :param locked: Indicates whether the create should run in a locked context (e.g. to prevent port conflicts)
-        :param plugins: Plugins that should be added to the configuration file
-        :param base_dir: Base directory that should contain the data and tlogs
-        :param ip: IP address of the first node of the new cluster
+        Always creates a cluster but marks it's usage according to the internal flag
         :param cluster_name: Name of the cluster
+        :type cluster_name: str
+
+        :param cluster_type: Type of the cluster (See ServiceType.ARAKOON_CLUSTER_TYPES)
+        :type cluster_type: str
+
+        :param ip: IP address of the first node of the new cluster
+        :type ip: str
+
+        :param base_dir: Base directory that should contain the data and tlogs
+        :type base_dir: str
+
+        :param plugins: Plugins that should be added to the configuration file
+        :type plugins: list
+
+        :param locked: Indicates whether the create should run in a locked context (e.g. to prevent port conflicts)
+        :type locked: bool
+
+        :param internal: Is cluster internally managed by OVS
+        :type internal: bool
+
+        :return: Ports used by arakoon cluster
+        :rtype: dict
         """
+        if cluster_type not in ServiceType.ARAKOON_CLUSTER_TYPES:
+            raise ValueError('Cluster type {0} is not supported. Please choose from {1}'.format(cluster_type, ', '.join(ServiceType.ARAKOON_CLUSTER_TYPES)))
+
         logger.debug('Creating cluster {0} on {1}'.format(cluster_name, ip))
         base_dir = base_dir.rstrip('/')
 
-        client = SSHClient(ip)
+        EtcdConfiguration.set('/ovs/framework/stores', {'persistent': 'pyrakoon',
+                                                        'volatile': 'memcache'})
```

Comment:
It doesn't make sense to overwrite the volatile configuration when you set up a new cluster.